api/resources_portal/test/views/test_organization_grant.py

```diff
@@ -137,6 +137,37 @@ def test_delete_request_disassociates_a_material(self):
         # Verify that the grant was not deleted, just its relationship
         Grant.objects.get(pk=self.grant.id)
 
+    def test_delete_request_transfers_materials(self):
+        grant = self.grant2
+        user = grant.user
+
+        # Add grant2 materials to organization1, so they can be transferred out.
+        materials = grant.materials.all()
+        material1 = materials[0]
+        material2 = materials[1]
+        material1.organization = self.organization1
+        material1.save()
+        material2.organization = self.organization1
+        material2.save()
+
+        # Create a personal organization for grant2.user (factory
+        # doesn't create by default.)
+        user.personal_organization = PersonalOrganizationFactory(owner=user)
+        user.save()
+
+        self.client.force_authenticate(user)
+        url = reverse("organizations-grants-detail", args=[self.organization1.id, grant.id])
+        response = self.client.delete(url + "?transfer=true")
+        self.assertEqual(response.status_code, 204)
+
+        grant.refresh_from_db()
+        self.assertNotIn(grant, self.organization1.grants.all())
+        # Verify that the grant was not deleted, just its relationship
+        grant = Grant.objects.get(pk=grant.id)
+
+        for material in grant.materials.all():
+            self.assertEqual(user.personal_organization, material.organization)
+
     def test_delete_fails_if_not_grant_owner(self):
         self.client.force_authenticate(user=self.organization1.owner)
```

Comment:
```suggestion
        self.client.force_authenticate(user=self.organization2.owner)
```

Reply:
the user here is from org1 and so is the grant, we could do this or add the grant to org2